### PR TITLE
Major Performance + Misc UIkit Improvements

### PR DESCRIPTION
**Pull request by sghoweri**
OK! 

In a nutshell, this PR is meant to start laying down a lot of the much needed groundwork for much larger and more ambitious PL changes down the road (think full on rewrite / refactor of UIkit's JavaScript, getting rid of Bower, etc) while still providing the Pattern Lab community a MAJOR speed bump we can all immediately benefit from in the interim.

Specifically, this PR aims to tackle a couple things:

1. Adjusts how all of Pattern Lab's CSS and JS load up to be as non-render blocking and async-friendly as possible (short of a full on component refactor / rewrite + move to more modern JS tooling like Webpack or Rollup -- saving that for a separate PR 😉)
    - Auto-generate Critical CSS via [Penthouse](https://github.com/pocketjoso/penthouse); wire up to Gulp
    - Async load / preload CSS via the Filament Group's [LoadCSS library](https://www.npmjs.com/package/fg-loadcss)
    - Async load JS when the async / defer-loaded JavaScript is ready to go (aka how to async load a bunch of old JavaScript written to be loaded synchronously without rewriting everything to use proper modern dependency management)
    - Automatically inline critical JS and CSS assets via the existing Gulp build pipeline (again, a lot of workarounds till the dependency-managed component refactor PR gets in)

### Before
screenshot_7_22_18__9_21_pm

### After
pattern_lab

> Note: ^ these local perf tests aren't on HTTP2 or with gzip / text compression so I'd expect even better metrics in the real world!

2. Adds a new loader UI to the empty iframe container to help display a little bit of content before Pattern Lab's JavaScript fully kicks in:

patternlab-loading-screen

3. Fixes up a couple misc UI bugs encountered putting together all this -- namely the viewport itself being off-center when the sidebar layout is switched on

4. Finally, this PR also preps us to be able to go full on Progressive Web App (via service workers) using [Google's Workbox project](https://developers.google.com/web/tools/workbox/) -- build tasks for this as commented out for the time being (needs some cache busting logic before 100% fully baked) but this gets the ball rolling to ultimately make PL one of the fastest Design System tools out there.

### CODE REVIEW

**Comment by bradfrost** [comment]
@sghoweri lots of great stuff here! Looking forward to diving into it more.

File: packages/uikit-workshop/penthouse.js [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    width: 1300, // viewport width
+    height: 900, // viewport height
+
+    // when true, will not filter out larger media queries
+    keepLargerMediaQueries: true,
+
+    // selectors to keep
+    forceInclude: [
+      // '.my-selector',
+      '.pl-c-body--theme-light',
```
Comment (bmuenzenmeyer) [inline packages/uikit-workshop/penthouse.js]: should all the theme classes be considered? or are they picked up by already being the default? That would make sense

**Comment by bmuenzenmeyer** [comment]
Manually resolved conflicts - looked pretty simple

File: packages/uikit-workshop/penthouse.js (hunk 1 above)
Comment (sghoweri) [reply packages/uikit-workshop/penthouse.js]: The light theme was the one variation Penthouse wasn’t immediately picking up when calculating out the Critical CSS for the page.

Ideally we’d just have these config options be exported in one of our JS files and the build step here pulls those in automatically (similar to how we deal with a ton of these gotchas in the Bolt Design System code).

Basically, if we go in and update which theme options are available, our build process stays up to date!

**Pull request merged**